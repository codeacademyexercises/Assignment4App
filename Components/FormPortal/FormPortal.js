import React, { Component } from 'react';
import {
  Text, View,
} from 'react-native';
import PropTypes from 'prop-types';
import styles from './FormPortal.style';

export default class FormPortal extends Component {
  render() {
    let display = '';
    const { FormName } = this.props;
    const { createdOn } = this.props;
    if (FormName.length > 30) {
      display = `${FormName.slice(0, 31)}...`;
    } else {
      display = FormName;
    }
    return (
      <View style={styles.FormNameContainer}>
        <Text style={styles.FormName}>
          {display}
        </Text>
        <View style={styles.horizontalLine} />
        <Text style={styles.createOn}>
                  Created On:
          {' '}
          {createdOn.slice(0, 10).replace(/[-]/g, '/')}
        </Text>
      </View>
    );
  }
}

FormPortal.propTypes = {
  FormName: PropTypes.string.isRequired,
  createdOn: PropTypes.string.isRequired,
};
