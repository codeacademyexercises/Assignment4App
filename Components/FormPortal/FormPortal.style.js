import {
  StyleSheet,
} from 'react-native';

export default StyleSheet.create({
  FormNameContainer: {
    paddingTop: 25,
    padding: 10,
    marginTop: 25,
    height: 130,
    width: 270,
    backgroundColor: 'white',
    alignItems: 'flex-start',
  },
  FormName: {
    height: 70,
    fontSize: 20,
  },
  horizontalLine: {
    width: 250,
    borderWidth: 1,
    borderColor: 'black',
  },
  createOn: {
    // marginLeft: 80,
    // marginTop: 5,
    alignSelf: 'flex-end',
  },
});
