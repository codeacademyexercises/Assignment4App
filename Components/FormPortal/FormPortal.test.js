import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FormPortal from './FormPortal';

configure({ adapter: new Adapter() });

describe('<Header /> ', () => {
  it('snapshot test', () => {
    const tree = renderer.create(<FormPortal FormName="Personal Details" createdOn="2019-02-27T06:47:39.504Z" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
