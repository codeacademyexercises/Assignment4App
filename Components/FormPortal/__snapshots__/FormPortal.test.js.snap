// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header />  snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "white",
      "height": 130,
      "marginTop": 25,
      "padding": 10,
      "paddingTop": 25,
      "width": 270,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "height": 70,
      }
    }
  >
    Personal Details
  </Text>
  <View
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 1,
        "width": 250,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "flex-end",
      }
    }
  >
    Created On: 
    2019/02/27
  </Text>
</View>
`;
